src/common/middleware/call-api.js

```diff
@@ -0,0 +1,101 @@
+import { checkStatus, getError } from '../helpers/api';
+
+// Action key that carries API call info interpreted by this Redux middleware.
+export const CALL_API = 'CALL_API';
+
+// A Redux middleware that interprets actions with CALL_API info specified.
+// Performs the call and promises when such actions are dispatched.
+export default defaults => () => next => async action => {
```

Comment:
Personally I find such nested function one-liners a bit hard to read especially when params are not wrapped in parens. I really don't know where one function starts and where other ends.

I know that using `return` for each single nested function would be a bit of boilerplate and indent, so it's not perfect too.

I'm not strongly against current state, but maybe it could be made more readable.